airflow/www/views.py

```diff
@@ -740,13 +742,12 @@ def index(self):
                 current_dags = all_dags
                 num_of_all_dags = all_dags_count
 
-            dags = (
-                current_dags.order_by(DagModel.dag_id)
-                .options(joinedload(DagModel.tags))
-                .offset(start)
-                .limit(dags_per_page)
-                .all()
+            sort_column = DagModel.__table__.c[arg_sorting_key]
+            current_dags = current_dags.order_by(
+                sort_column.is_(None), desc(sort_column) if arg_sorting_direction == "desc" else sort_column
```

Comment:
This is too cramped. Please split the logic into multiple statements.

Reply:
Done :)